### PR TITLE
DlgMakeDate

### PR DESCRIPTION
**Pull request by stevenndungu**
In progress

### CODE REVIEW

**Comment by stevenndungu** [comment]
Two things remaining - setting passing of parameters correctly on reset (under rdosingle)
                                    -setting rdosingle and rdodefault as default rdo bittons checked

**Comment by Lunalo** [comment]
This is not ready for review since we are still looking into something .

**Comment by stevenndungu** [comment]
@africanmathsinitiative/developers this is ready for review.

**Comment by Lunalo** [comment]
The dialog is working really fine. but as we have discussed this morning , Can we have an easier way of giving the format? @dannyparsons , @rdstern . I am not sure if someone new to r will know how to type something like "%B/%d/%y"

**Comment by Lunalo** [comment]
Tab order needs to be set

**Comment by dannyparsons** [comment]
Yes, there needs to be much more flexibility in giving the format. However, that's probably a separate discussion the dialog overall, but it does need to be improved soon.

**Comment by Lunalo** [comment]
@dannyparsons 

This is ready for you

File: instat/dlgMakeDate.vb [hunk 1]
```diff
@@ -13,144 +13,278 @@
...
+        ucrInputDayOption.SetItems(dctDayItems)
+
+        ucrInputComboBoxMonthTwo.SetParameter(New RParameter("doy_typical_length", 3))
+        Dim dctMonthTwoItems As New Dictionary(Of String, String)
+        dctMonthTwoItems.Add("365/366", Chr(34) & "365/366" & Chr(34))
+        dctMonthTwoItems.Add("366", Chr(34) & "366" & Chr(34))
+        ucrInputComboBoxMonthTwo.SetItems(dctMonthTwoItems)
+
+        ucrInputFormat.SetParameter(New RParameter("format", 1))
+        Dim dctDateFormat As New Dictionary(Of String, String)
```
Comment (dannyparsons) [inline instat/dlgMakeDate.vb]: Your defining of variables should always be at the top of the sub.

**Comment by stevenndungu** [comment]
@dannyparsons made the changes suggested

**Comment by dannyparsons** [comment]
@lunalo could you have a final test of everything here?

**Comment by Lunalo** [comment]
@dannyparsons 

This is in a good shape for now

**Pull request merged**